net/libfab/libfab.c

```diff
@@ -86,30 +88,29 @@ static int libfab_dom_init(struct m0_net_xprt *xprt, struct m0_net_domain *dom)
 	if (fab_dom == NULL)
 		return M0_RC(-ENOMEM);
 
-	fab_dom->fab_hints = fi_allocinfo();
-	if (fab_dom->fab_hints != NULL) {
+	fab_dom->fdp_hints = fi_allocinfo();
+	if (fab_dom->fdp_hints != NULL) {
 		/*
 		* TODO: Added for future use
-		* fab_dom->fab_hints->ep_attr->type = FI_EP_RDM;
-		* fab_dom->fab_hints->caps = FI_MSG;
-		* fab_dom->fab_hints->fabric_attr->prov_name = "verbs";
+		* fab_dom->fdp_hints->ep_attr->type = FI_EP_RDM;
+		* fab_dom->fdp_hints->caps = FI_MSG;
+		* fab_dom->fdp_hints->fabric_attr->prov_name = "verbs";
 		*/
 		rc = fi_getinfo(FI_VERSION(FI_MAJOR_VERSION,FI_MINOR_VERSION),
-				NULL, NULL, 0, fab_dom->fab_hints,
-				&fab_dom->fab_fi);
+				NULL, NULL, 0, fab_dom->fdp_hints,
+				&fab_dom->fdp_fi);
 		if (rc == FI_SUCCESS) {
-			rc = fi_fabric(fab_dom->fab_fi->fabric_attr,
-				       &fab_dom->fab_fabric, NULL);
+			rc = fi_fabric(fab_dom->fdp_fi->fabric_attr,
+				       &fab_dom->fdp_fabric, NULL);
 			if (rc == FI_SUCCESS) {
-				rc = fi_domain(fab_dom->fab_fabric,
-					       fab_dom->fab_fi,
-					       &fab_dom->fab_domain, NULL);
+				rc = fi_domain(fab_dom->fdp_fabric,
+					       fab_dom->fdp_fi,
+					       &fab_dom->fdp_domain, NULL);
 				if (rc == FI_SUCCESS)
 					dom->nd_xprt_private = 
-							    fab_dom->fab_domain;
+							    fab_dom->fdp_domain;
 			}
 		}
-		fi_freeinfo(fab_dom->fab_hints);
```

Comment:
need to free hints at the end of dom_init

Reply:
And to cleanup all previous initializations.

Reply:
fi_freeinfo(fab_dom->fab_hints); is handled in dom_fini()
if we do need hits for future then we can remove fdp_hints from struct m0_fab__dom_param and keep local copy

Reply:
The point here is that:
If a function fails, especially, this kind of xxx_init() functions, it should cleanup all resources it creates and then return failure.

When libfab_dom_init() failes, libfab_dom_fini() will not be called.

Reply:
yes, we can remove hints from m0_fab__dom_param and use a local copy instead